### PR TITLE
Update API to latest main

### PR DESCRIPTION
**Pull request by bergundy**
## What changed?

- Also added propagate a couple of new fields to nexus requests.

### CODE REVIEW

**Review by stephanos — APPROVED** [review]

File: go.mod [hunk 1]
```diff
@@ -44,7 +44,7 @@ require (
 	go.opentelemetry.io/otel/metric v1.26.0
 	go.opentelemetry.io/otel/sdk v1.26.0
 	go.opentelemetry.io/otel/sdk/metric v1.26.0
-	go.temporal.io/api v1.32.1-0.20240506171507-d8690dde1d05
+	go.temporal.io/api v1.32.1-0.20240509192021-ccbeafbf4886
```
Comment (dnr) [inline go.mod:47]: Are you going to do a release of api?

Comment (bergundy) [reply go.mod:47]: I can.

Comment (bergundy) [reply go.mod]: Just did and update `go.mod`.

**Review by alexshtin — APPROVED** [review]
Please merge is asap. `main` is broken.

**Pull request merged**